providers/oidc_test.go

```diff
@@ -434,23 +504,90 @@ func TestOIDCProvider_EnrichSession(t *testing.T) {
 			ProfileJSON: map[string]interface{}{
 				"email": "new@thing.com",
 			},
+			IntrospectJSON: map[string]interface{}{
+				"active": true,
+			},
 			ExpectedError: nil,
 			ExpectedSession: &sessions.SessionState{
+				User:             "already",
+				Email:            "already@populated.com",
+				IDToken:          idToken,
+				AccessToken:      accessToken,
+				RefreshToken:     refreshToken,
+				IntrospectClaims: "eyJhY3RpdmUiOnRydWV9",
+			},
+		},
+		"Introspection Response added in session": {
+			ExistingSession: &sessions.SessionState{
 				User:         "already",
 				Email:        "already@populated.com",
 				IDToken:      idToken,
 				AccessToken:  accessToken,
 				RefreshToken: refreshToken,
 			},
+			EmailClaim:  "email",
+			GroupsClaim: "groups",
+			ProfileJSON: map[string]interface{}{
+				"email": "new@thing.com",
+			},
+			IntrospectJSON: map[string]interface{}{
```

Comment:
This looks an awful lot like an ID Token payload, does this normally contain a lot more information? Do you know which common providers offer the introspection endpoint?

Reply:
Yes this would normally contain a lot more info but would not be in any fixed format.

The only major provider I could find with direct support was [Okta](https://developer.okta.com/docs/reference/api/oidc/#introspect)